tests/test_virtualenv.py

```diff
@@ -279,6 +279,66 @@ def test__resolved_interpreter_windows_pyexe_fails(sysfind, make_one):
     sysfind.assert_any_call("py")
 
 
+@pytest.mark.parametrize(
+    ["input_", "path_result", "expected"],
+    [
+        ("3.7", r"c:\python37-x64\python.exe", r"c:\python37-x64\python.exe"),
+        ("3.7", r"c:\fails\python.exe", None),
+        ("python3.7", r"c:\python37-x64\python.exe", None),
+        ("2.7", r"c:\python37-x64\python.exe", None),
+        ("goofy", r"c:\python37-x64\python.exe", None),
+        ("3.7", None, None),
+        ("python3.7", None, None),
+        ("2.7", None, None),
+        ("goofy", None, None),
+    ],
+)
+@mock.patch("nox.virtualenv._SYSTEM", new="Windows")
+@mock.patch.object(py.path.local, "sysfind")
+def test__resolved_interpreter_windows_path_and_version(
+    sysfind, make_one, input_, path_result, expected
+):
+    # Establish that if we get a standard pythonX.Y path, we look it
+    # up via the path on Windows.
+    venv, _ = make_one(interpreter=input_)
+
+    # Trick the system into thinking that it cannot find python3.6
+    # (it likely will on Unix). Also, we don't give it a dummy
+    # py launcher. But we give it a dummy python interpreter to find
+    # in the system path.
+    if path_result:
+
+        def mock_sysexec(_):
+            if path_result == r"c:\fails\python.exe":
+                raise py.process.cmdexec.Error(1, 1, "", "", "")
+            else:
+                return "Python 3.7.3 | Anaconda\\n"
+
+        attrs = {
+            "sysexec.side_effect": mock_sysexec,
+            "__str__.return_value": path_result,
+        }
+        mock_python = mock.MagicMock()
```

Comment:
1. `MagicMock` needed over `Mock` because of `__str__`?
1. Any way we can reduce the surface area by using `MagicMock(spec=())`?

Reply:
I did not get "__str__" to work using `Mock`, but did with `MagicMock`, and looking at its [documentation](https://docs.python.org/3/library/unittest.mock.html#unittest.mock.MagicMock), an empty `spec` should just make it a `Mock`. I will look into getting plain `Mock` to work again, otherwise maybe resort to `MagicMock(spec=("__str__"))`.

Disclaimer: First-time-user of unittest.mock ;)

Reply:
SGTM. Bravo on diving in!